### PR TITLE
Indexer service

### PR DESCRIPTION
**Pull request by miiu96**
#### Description of the pull request (what is new / what has changed)

Documentation how to use the `elasticindexer` service

#### Did you test the changes locally ?
- [x] yes
- [ ] no

#### Which category (categories) does this pull request belong to?
- [x] document new feature
- [ ] update documentation that is not relevant anymore
- [ ] add examples or more information about a component
- [ ] fix grammar issues
- [ ] other

### CODE REVIEW

File: docs/sdk-and-tools/elastic-search.md [hunk 1]
```diff
@@ -19,6 +19,21 @@ Our implementation uses a concept of a queue and makes sure that everything is b
...
+
+Set up four observers, each corresponding to a shard, with the `WebSocketHost` enabled, and create an instance of the `elasticindexer` service.
+Follow the instructions provided on [this](/sdk-and-tools/indexer) page to accomplish this.
+
+[comment]: # (mx-context-auto)
+
+### Option 2 
+
+:::warning
+this Option will be removed on the release `rc/v1.7.0`
```
Comment (iulianpascalau) [inline docs/sdk-and-tools/elastic-search.md:34]: we are not really sure about this now, maybe state this as:
```
we plan to remove this option in the release `rc/v1.7.0`
```

File: docs/sdk-and-tools/indexer.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+[comment]: # (mx-context-auto)
+
+### Observer Client
+
+On the observer side, there is a WebSocket host that will send messages to the elasticindexer service.
+
+In the observer node's configuration directory, `external.toml` config file can be configured
+to enable host driver config. The config file can be found
+[here](https://github.com/multiversx/mx-chain-go/blob/master/cmd/node/config/external.toml).
```
Comment (iulianpascalau) [inline docs/sdk-and-tools/indexer.md:42]: this link does not work (yet):
use this one?
https://github.com/multiversx/mx-chain-go/blob/rc/v1.6.0/cmd/node/config/external.toml

File: docs/sdk-and-tools/indexer.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
+---
+id: indexer
+title: Elasticindexer service
+---
+
+[comment]: # (mx-abstract)
+
+## Overview
+
+A MultiversX observer node can send messages over `WebSocket` to an elastic indexer service, which will process and index the data in an Elasticsearch database.
```
Comment (iulianpascalau) [inline docs/sdk-and-tools/indexer.md:10]: add a statement here that this feature will work starting from rc/v1.6.0 version of the node?

File: docs/sdk-and-tools/indexer.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        retry-duration-in-seconds = 5
+        # Signals if in case of data payload processing error, we should send the ack signal or not
+        blocking-ack-on-error = true
+        # After a message will be sent it will wait for an ack message if this flag is enabled
+        with-acknowledge = true
+        # The duration in seconds to wait for an acknowledgment message, after this time passes an error will be returned
+        acknowledge-timeout-in-seconds = 50
+```
+
+The supported config variables for the `elastic-cluster` section:
```
Comment (mariusmihaic) [inline docs/sdk-and-tools/indexer.md:121]: Same comment

**Review by mariusmihaic — COMMENTED** [review]
+ what @iulianpascalau mentioned

File: docs/sdk-and-tools/elastic-search.md (hunk 1 above)
Comment (miiu96) [reply docs/sdk-and-tools/elastic-search.md:34]: changed

File: docs/sdk-and-tools/indexer.md (hunk 2 above)
Comment (miiu96) [reply docs/sdk-and-tools/indexer.md:42]: changed the link

File: docs/sdk-and-tools/indexer.md (hunk 3 above)
Comment (miiu96) [reply docs/sdk-and-tools/indexer.md:10]: added

File: docs/sdk-and-tools/indexer.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+On the observer side, there is a WebSocket host that will send messages to the elasticindexer service.
+
+In the observer node's configuration directory, `external.toml` config file can be configured
+to enable host driver config. The config file can be found
+[here](https://github.com/multiversx/mx-chain-go/blob/master/cmd/node/config/external.toml).
+
+The supported config variables are as follows:
+
+- `Enabled`: signals whether a driver should be attached when launching the node.
```
[the remark this answers is not in the document]
Comment (miiu96) [reply docs/sdk-and-tools/indexer.md:46]: removed

File: docs/sdk-and-tools/indexer.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+```
+
+[comment]: # (mx-context-auto)
+
+### Elasticindexer service
+
+In the `elasticindexer` configuration directory (`cmd/elasticindexer/config`), there is the `prefs.toml`
+file that can be used to configure the service.
+
+The supported config variables for web-socket section are:
```
[the remark this answers is not in the document]
Comment (miiu96) [reply docs/sdk-and-tools/indexer.md:88]: removed

File: docs/sdk-and-tools/indexer.md (hunk 4 above)
Comment (miiu96) [reply docs/sdk-and-tools/indexer.md]: removed

**Review by iulianpascalau — APPROVED** [review]

**Review by mariusmihaic — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**